osmaxx/excerptexport/views.py

```diff
@@ -69,31 +73,91 @@ def get_object(self, queryset=None):
         return o
 
 
-class ExportsListView(LoginRequiredMixin, FrontendAccessRequiredMixin, ListView):
+class ExportsListMixin:
+    @property
+    def excerpt_ids(self):
+        if not hasattr(self, '_excerpt_ids'):
+            self._excerpt_ids = list(
+                OrderedSet(
+                    self.get_user_exports()
+                    .values_list('extraction_order__excerpt', flat=True)
+                )
+            )
+        return self._excerpt_ids
+
+    @property
+    def status_choices(self):
+        allowed_choices = [
+            statuses.FINISHED,
+            statuses.FAILED,
+        ]
+        return [choice for choice in Export.STATUS_CHOICES if choice[0] in allowed_choices]
+
+    def get_user_exports(self):
+        return self._filter_exports(
+            Export.objects.filter(extraction_order__orderer=self.request.user)
+        ).order_by('-updated', '-finished')
+
+    def _filter_exports(self, query):
+        allowed_status_filters = [
+            status[0] for status in self.status_choices
+        ]
+        status_filter = self.request.GET.get('status', None)
+        if status_filter and status_filter in allowed_status_filters:
+            return query.filter(status=status_filter)
+        return query
+
+    def _get_extra_context_data(self):
+        return dict(
+            status_choices=self.status_choices,
+            status_filter=self.request.GET.get('status', None),
+        )
+
+
+class ExportsListView(LoginRequiredMixin, FrontendAccessRequiredMixin, ExportsListMixin, ListView):
     template_name = 'excerptexport/export_list.html'
-    context_object_name = 'exports'
-    model = Export
-    ordering = ['-extraction_order__excerpt', '-status']
+    context_object_name = 'excerpts'
+    model = Excerpt
+    paginate_by = 5
+
+    def get_context_data(self, **kwargs):
+        context_data = super().get_context_data(**kwargs)
+        context_data.update(self._get_extra_context_data())
+        context_data['excerpt_list_with_exports'] = OrderedDict(
+            (
+                excerpt,
+                self.get_user_exports().filter(extraction_order__excerpt__pk=excerpt.id)
+                .select_related('extraction_order', 'extraction_order__excerpt', 'output_file'))
```

Comment:
consider extracting `self.get_user_exports().filter(extraction_order__excerpt__pk=excerpt.id).select_related('extraction_order', 'extraction_order__excerpt', 'output_file')` to a function, so that the pairs (2-tuples) inside the generator expression become more apparent for the OrderedDict initialization without having to spell them out as `tuple(..., ...)`.